Add new references for cell type annotation

allyhawkins opened the pull request:
Closes #416 

This PR adds additional references to use with `CellAssign`, using the organs available in the PanglaoDB marker gene list. 

- I included all references we may need to run each project (see https://github.com/AlexsLemonade/ScPCA-admin/pull/713). This includes two references that combine organs: 
  - `adrenal-reproductive-compartment`, which includes the adrenal gland, pancreas, and reproductive tissue to account for the project that contains both Adrenal corticocarcinoma and a germ cell tumor. 
  - `bone-and-soft-tissue-compartment`, which includes bone, smooth muscle, and connective tissue to account for the projects that contain samples that come from various bone and soft tissue areas in the body. 
 - I also added some extra references for various organs that we might want to use by themselves at some point, like `lung-compartment` and `liver-compartment`. 
 - The last thing I did was add `vasculature` to the list of organs used to build each reference. We want to include immune cells in each reference, but it would also be important to include anything related to blood vessels that might be in tumors. Specifically, the `vasculature` organ in the database only contains endothelial cells (and some subtypes of endothelial cells. These are commonly found in solid tumors, so it's important to include them and err on the side of building a broader reference. 

I'm going to request two reviewers here, because it's important to get extra eyes on the decisions regarding which organs go in each reference and the names that I chose.

allyhawkins commented:
Also I haven't actually run the workflow yet. Once we agree on the names/ organs then I will generate the references.

sjspielman commented on references/celltype-reference-metadata.tsv:
```diff
@@ -3,6 +3,16 @@ BlueprintEncodeData	celldex	SingleR	NA
+bone-and-soft-tissue-compartment	PanglaoDB	CellAssign	Bone, Smooth muscle, Connective tissue, Immune system, Vasculature
+reproductive-compartment	PanglaoDB	CellAssign	Reproductive, Immune system, Vasculature
+pancreas-compartment	PanglaoDB	CellAssign	Pancreas, Immune system, Vasculature
+lung-compartment	PanglaoDB	CellAssign	Lungs, Immune system, Vasculature
```
and `Adrenal glandS`.. harrumph.

allyhawkins replied:
I agree, this is silly.

sjspielman reviewed (commented):
Nothing really jumped out at me that I _disagree_ with here, except for one of the names that I can't come up with anything else for!

jashapiro reviewed (commented):
This looks like a good start! 

I had a few thoughts/questions
- Alphabetize? (Just for future ease)
- Do we want connective tissue in more of the organs? Not sure how exactly that breaks down, but I imagine we would want it in the kidney & pancreas, for example to get the fibroblasts.
- A shorter name for "bone and soft tissue"? I don't have a great thought here, unfortunately. If this was meant as a "catch-all" do we want to include even more in it?

allyhawkins commented on references/celltype-reference-metadata.tsv:
```diff
@@ -3,6 +3,16 @@ BlueprintEncodeData	celldex	SingleR	NA
-blood-compartment	PanglaoDB	CellAssign	Blood, Bone, Immune system
-muscle-compartment	PanglaoDB	CellAssign	Connective tissue, Immune system, Skeletal muscle, Smooth muscle
-brain-compartment	PanglaoDB	CellAssign	Brain, Immune system
+blood-compartment	PanglaoDB	CellAssign	Blood, Bone, Immune system, Vasculature
```
You have a good point, I think I would just remove them all together here.

allyhawkins commented on references/celltype-reference-metadata.tsv:
```diff
@@ -3,6 +3,16 @@ BlueprintEncodeData	celldex	SingleR	NA
+adrenal-compartment	PanglaoDB	CellAssign	Adrenal glands, Connective tissue, Immune system, Vasculature
+kidney-compartment	PanglaoDB	CellAssign	Kidney, Immune system, Vasculature
+eye-compartment	PanglaoDB	CellAssign	Eye, Immune system, Vasculature
+adrenal-reproductive-compartment	PanglaoDB	CellAssign	Adrenal glands,  Connective tissue, Smooth muscle, Pancreas, Reproductive, Immune system, Vasculature
```
I was also conflicted about this, what about `endocrine-reproductive-compartment`?

sjspielman replied:
I honestly don't know, I feel like if we go full `endocrine` we'd want more in there like thymus, etc. That said, I feel like `reproductive` is part of `endocrine`, and just `endocrine-compartment` _could_ be an option? Since such broad terms are ambiguous though, and for everything else we refer to exact organs, I might actually take it back and avoid `endocrine`? Still tricky!!

Of course there's always the `grabbag-compartment` option 🙃

allyhawkins replied:
We could just be explicit about what's in there and use `adrenal-pancreas-reproductive-compartment`. ?

jashapiro replied:
I think for multiple organ systems, we could probably drop `compartment` and use `adrenal-pancreas-reproductive`? or maybe `endocrine-system` and we do include thymus?

allyhawkins replied:
Also the reason `smooth muscle` is in here is because that project has an "abdominal mass" too. But its an ACC tumor in the abdomen.

jashapiro replied:
Or we could go with `abdominal-organs` and include GI tract? (then leave off thymus?)

jashapiro commented:
Do we also maybe want to have available `all-organs` or something like that?

allyhawkins commented:
> Alphabetize? (Just for future ease)

I went ahead and alphabetized by the ref name. 

> Do we want connective tissue in more of the organs? Not sure how exactly that breaks down, but I imagine we would want it in the kidney & pancreas, for example to get the fibroblasts.

Yes, I think this is a good idea. I added it to pancreas, lungs, liver, and bone. 

> A shorter name for "bone and soft tissue"? I don't have a great thought here, unfortunately. If this was meant as a "catch-all" do we want to include even more in it?

I don't know if this really should be a "catch-all". I think we should actually have one for bone and soft tissue diseases that doesn't have other organs in it. I did update the name to remove "compartment" and then also used the `adrenal-pancreas-reproductive` name that you suggested. 

However, because we have some strange projects, maybe we make a "general-pediatric-disease" reference. This could include any tissues that are likely found to be affected by peds tumors. I don't know about the name, but maybe we could make one that's more general? It would be really big, but at least it would have a lot of options?

allyhawkins commented:
> Do we also maybe want to have available all-organs or something like that?

I did not see this comment while I was working on my last response, but it looks like we came up with similar ideas. All organs would be a lot and includes things like heart that is unlikely to have a tumor. But I think anyplace where a tumor could arise is fair game.

sjspielman commented on references/celltype-reference-metadata.tsv:
```diff
@@ -1,8 +1,18 @@
 celltype_ref_name	celltype_ref_source	celltype_method	organs
+adrenal-compartment	PanglaoDB	CellAssign	Adrenal glands, Connective tissue, Immune system, Vasculature
+adrenal-pancreas-reproductive	PanglaoDB	CellAssign	Adrenal glands,  Connective tissue, Smooth muscle, Pancreas, Reproductive, Immune system, Vasculature
```
I think this is a good choice since it really does combine our adrenal, pancreas, and reproductive references

sjspielman commented:
Just commenting with all the organs for reference (pun?!) - 

```
> unique(dat$organ)
 [1] "Pancreas"           "Connective tissue"  "Brain"             
 [4] "Lungs"              "Smooth muscle"      "Immune system"     
 [7] "Epithelium"         "Heart"              "Liver"             
[10] "Adrenal glands"     "GI tract"           "Reproductive"      
[13] "Kidney"             "Zygote"             "Vasculature"       
[16] "Embryo"             "Blood"              "Thyroid"           
[19] NA                   "Bone"               "Skin"              
[22] "Mammary gland"      "Eye"                "Skeletal muscle"   
[25] "Olfactory system"   "Parathyroid glands" "Oral cavity"       
[28] "Thymus"             "Placenta"           "Urinary bladder"   
```

allyhawkins commented:
I think if we want to do all organs, then I would maybe include everything, but `NA`. But if we are going to do a cancer-specific reference, then I would use the following organs: 

 [1] "Pancreas"           "Connective tissue"  "Brain"             
 [4] "Lungs"              "Smooth muscle"      "Immune system"     
 [7] "Epithelium"         ~~"Heart"~~              "Liver"             
[10] "Adrenal glands"     "GI tract"           "Reproductive"      
[13] "Kidney"             "Zygote"             "Vasculature"       
[16] "Embryo"             "Blood"              "Thyroid"           
[19] ~~NA~~                   "Bone"               "Skin"              
[22] "Mammary gland"      "Eye"                "Skeletal muscle"   
[25] ~~"Olfactory system"~~   "Parathyroid glands" "Oral cavity"       
[28] "Thymus"             ~~"Placenta"~~          "Urinary bladder"   

Now if we are talking about pediatric cancer specific, I would remove `Mammary gland` too.

sjspielman commented:
What about zygote and embryo? Would those be relevant for pediatric cancer? Definitely deferring to all @allyhawkins's expertise for all this!

jashapiro commented:
I would leave `Olfactory system` in an "all cancer" reference, but I can concur about removing it and `Mammary gland` from from pediatric cancer.

allyhawkins commented:
> What about zygote and embryo? Would those be relevant for pediatric cancer? Definitely deferring to all @allyhawkins's expertise for all this!

I was on the fence with these... 
For zygote, the only cell type they have is `Embryonic stem cells`, so I think it's important to keep those. A lot of pediatric solid tumors are derived from ESC's and show similar characteristics. 

For embryo, the cell types that they have `epiblast`, `trophoblast`, etc are cell states that are precursors to mesoderm/ectoderm which is also where a handful of pediatric tumors originate. I feel like I would keep them in because identifying tumor cells that are similar to those cells can be helpful information.

allyhawkins commented:
do we want a `all-cancer` and a `pediatric-cancer` reference?

jashapiro commented:
> do we want a `all-cancer` and a `pediatric-cancer` reference?

Why not? I don't know whether we would also want an `adult-cancer` that doesn't include zygote/embryo, but I think it probably makes sense to include those anyway, which would make it kind of redundant.

sjspielman commented:
> >  What about zygote and embryo? Would those be relevant for pediatric cancer? Definitely deferring to all @allyhawkins's expertise for all this!
> 
> I was on the fence with these...
> For zygote, the only cell type they have is Embryonic stem cells, so I think it's important to keep those. A lot of pediatric solid tumors are derived from ESC's and show similar characteristics.
> 
> For embryo, the cell types that they have epiblast, trophoblast, etc are cell states that are precursors to mesoderm/ectoderm which is also where a handful of pediatric tumors originate. I feel like I would keep them in because identifying tumor cells that are similar to those cells can be helpful information.

Yeah, this is what I was wondering - if pediatric tumors probably _originate_ from similar tissues even if they don't formally "have" those tissues themselves. So, this makes sense to me to keep them.

jashapiro commented:
> For zygote, the only cell type they have is `Embryonic stem cells`, so I think it's important to keep those. A lot of pediatric solid tumors are derived from ESC's and show similar characteristics.

Does that mean we want to add those to all `-compartments`?

allyhawkins commented:
> > For zygote, the only cell type they have is `Embryonic stem cells`, so I think it's important to keep those. A lot of pediatric solid tumors are derived from ESC's and show similar characteristics.
> 
> Does that mean we want to add those to all `-compartments`?

I think most of the "embryonal" tumors are in the brain/CNS and then some sarcomas and solid tumors. So it's probably worth adding to any of the "solid tumor" `-compartments`. Leukemias and blood disorders arise from HSCs, so I don't think we would need it in the `blood-compartment`. 

Also for reference, from a [review that I found](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6877479/): 

> Embryonal tumors (ETs) can be divided into ETs of the central nervous system (CNS) in infants, such as medulloblastoma, medulloepitheliomas, atypical teratoid/rhabdoid tumors, primitive neuroectodermal tumors (PNETs) that occur in brain and spinal cord, neuroblastoma of CNS, embryonal tumors with multilayered rosettes, and other ETs or childhood blastomas that are derived from embryonal or fetal stem cells, such as neuroblastoma, retinoblastoma, rhabdomyosarcoma, nephroblastoma-Wilms' Tumor, ependymoma, and hepatoblastoma ([1](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6877479/#B1)).

allyhawkins commented:
Okay I went ahead and did some rearranging based on this discussion: 

- I added an `all-cancers` and `pediatric-cancers` reference. This includes all of the organs we discussed above and the difference between the two is the removal of `mammary gland` and `olfactory system` for the `pediatric-cancers` reference. 
- I added `Embryo` to _most_ of the existing `-compartments` references, with the exception of `bone` and `blood`. I did include it in `lungs`, `pancreas`, and `reproductive` although I'm not sure they are entirely relevant there. So I'm curious what others think. I think I would keep `lungs` because it's a prime spot for mets of many of the embryonal tumors, but I wasn't sure about the other two. 
- I also alphabetized the lists of `organs` in that column.

jashapiro reviewed (approved):
Looks good, and I like where this ended up, but I would just order the table first by reference source, then reference name.
